lib/lf/geometry/geometry_interface.h

```diff
@@ -125,6 +129,43 @@ class Geometry {
    */
   virtual std::unique_ptr<Geometry> SubGeometry(dim_t codim, dim_t i) const = 0;
 
+  /**
+   * @brief Generate a geometry object arising in the course of refinement
+   * 
+   * @param ref_pattern A code indicating the particular refinement of the element
+   * @paream selector A number selecting a particular child element
+   *
+   * Implementation of local mesh refinement entails splitting of elements into
+   * parts ("children"), whose shape will depend on the refinement pattern. 
+   * This method creates the geometry objects describing the shape of children.
+   *
+   * The arguments are unspecified integers to be interpreted correctly by
+   * specializations of this method. At the time of the definition of the interface
+   * all possible refinement patterns cannot be predicted. This is why  
+   * vanilla integer arguments have been chosen for this method.
+   */
+  virtual std::vector<std::unique_ptr<Geometry>>
+  ChildGeometry(const RefinementPattern &ref_pat) const {
+    LF_VERIFY_MSG(false,
+		  "ChildGeometry not implemented for " << typeid(*this).name());
+    return std::move(std::vector<std::unique_ptr<Geometry>>{});
```

Comment:
I'm not a big fan of this because
- strictly speaking it makes Geometry a non-[interface class](https://google.github.io/styleguide/cppguide.html#Interfaces)
- By looking at the code of a derived class it is not clear that this method is not implemented. Just the absence of the `ChildGeometry()` leads to a specific behavior. 

I know it's a little bit more work to implement this method also in all derived classes, but I think it's worth it because of code clarity. I have fixed this in one of my commits.